ch1-arraysAndStrings/saramic_javascript/eg_1_1_isUnique.spec.js

```diff
@@ -0,0 +1,44 @@
+const { isUnique } = require("./eg_1_1_isUnique");
+
+// 1.1 - Is Unique
+// Implement an algorithm to determine if a string has all unique characters.
+// What if you cannot use additional data structures?
+describe("IsUnique", () => {
+  it("empty string has unique characters", () => {
+    expect(isUnique("")).toBeTruthy();
+  });
+
+  it("alphabetical string has unique characters", () => {
+    expect(isUnique("abc")).toBeTruthy();
+  });
+
+  it("duplicate character does NOT have unique characters", () => {
+    expect(isUnique("aabc")).toBeFalsy();
+  });
+
+  it("mixed case alphabetical string has unique characters", () => {
+    expect(isUnique("aAbBcC")).toBeTruthy();
+  });
+
+  describe("multi width and similar characters", () => {
+    it("a string with unique emoji has unique characters", () => {
+      expect(isUnique("😀😃😄😁😆😅😂🤣")).toBeTruthy();
+    });
+
+    it("a string with duplicate same emoji does NOT have unique characters", () => {
+      expect(isUnique("😀😀😃")).toBeFalsy();
```

Comment:
These 3 emojis are not the same. You have 2 of [these](https://www.iemoji.com/view/emoji/885/smileys-people/grinning-face) and one of [these](https://www.iemoji.com/view/emoji/3/smileys-people/smiling-face-with-open-mouth)